### ISSUE TITLE
Feature: Local user

### ISSUE THREAD

**Issue by OgreTransporter** [tim-lebedkov/npackd-cpp#11]
I would like to add a flag to the build to set whether or not the program is looking for the known software. Users without admin privileges can't install/uninstall/modify software like .NET, Java, etc., so it's not necessary to manage that packages. With a second flag I want to set whether the program has to work with admin privileges or whether the local user is sufficient. Instead of build flags I can also add options to npacked. Instead of options it's also possible to detect current privileges (NetUserGetInfo) during program start and configure npacked with the available features. If somebody needs a special program it's still possible to run npacked with admin privileges.

I want to make a repository of ZIP archives available to my colleagues who all have no admin privileges. The programs are all portable and they run without admin rights. NET, Java etc. are managed by our administrator, so I don't need to search and display them.

Maybe it's also useful to add a new flag to packet definition which mark programs without admin privileges. So, depending on existing privileges it's possible to filter the list of available programs.

**Comment by tim-lebedkov** [comment]
This is generally a good idea. If you can implement it without a build flag, it would be great.

If possible, please merge in the UI the globally installed software with the software installed by the current user.

C:\Users\<user name>\AppData\Local\Npackd seems like an appropriate location for storing the package database.

Maybe it is better to just try to write to HKEY_LOCAL_MACHINE\SOFTWARE\Npackd\Npackd and use HKEY_CURRENT_USER\Software\Npackd\Npackd if it fails.

I will add a new repository at npackd.org for the applications that can be installed without admin permissions.

**Comment by OgreTransporter** [comment]
> This is generally a good idea. If you can implement it without a build flag, it would be great.

Ok, which kind? Detecting admin privileges by API (NetUserGetInfo) or user settings in npacked?

**Comment by OgreTransporter** [comment]
I've started a new branch for testing: https://github.com/OgreTransporter/npackd-cpp/commits/privileges

**Comment by tim-lebedkov** [comment]
I think that it's better to not expose this setting in Npackd user interface, but detect admin rights.

Just install a package globally if the user has admin rights and install the package locally otherwise. Maybe an option for NpackdCL would be useful. 

I also don't think it is necessary to use NetUserGetInfo . Just try to write to HKEY_LOCAL_MACHINE\SOFTWARE\Npackd\Npackd. If it is not possible, use it as a read-only list of installed packages.

Anyway, these are details. If you need to do something differently, please go on.

**Comment by OgreTransporter** [comment]
I think it's not enough to test HKEY_LOCAL_MACHINE for write access, but I found a method provided by MS to detect privileges: AccessCheck (https://msdn.microsoft.com/de-de/library/windows/desktop/aa374815(v=vs.85).aspx). There is also a sample from MS with this function (https://support.microsoft.com/en-us/help/118626/how-to-determine-whether-a-thread-is-running-in-user-context-of-local). I've created a static function (bool WPMUtils::hasAdminPrivileges()) which detects admin privileges.
